visualization/app/codeCharta/ui/codeMap/codeMap.actions.service.ts

```diff
@@ -48,34 +48,45 @@ export class CodeMapActionsService {
 		const { edges } = state.fileSettings
 		const { edgeMetric } = state.dynamicSettings
 		const numberOfEdgesToDisplay = state.appSettings.amountOfEdgePreviews
-		const edgePreviewNodes = this.edgeMetricDataService.getNodesWithHighestValue(edgeMetric, numberOfEdgesToDisplay)
-
-		edges.forEach(edge => {
-			if (
-				(edgePreviewNodes.includes(edge.fromNodeName) || edgePreviewNodes.includes(edge.toNodeName)) &&
-				Object.keys(edge.attributes).includes(edgeMetric)
-			) {
-				edge.visible = EdgeVisibility.both
-				if (!edgePreviewNodes.includes(edge.fromNodeName)) {
-					edge.visible = EdgeVisibility.to
-				} else if (!edgePreviewNodes.includes(edge.toNodeName)) {
-					edge.visible = EdgeVisibility.from
+		const edgePreviewNodes = new Set(this.edgeMetricDataService.getNodesWithHighestValue(edgeMetric, numberOfEdgesToDisplay))
+
+		for (const edge of edges) {
+			if (edge.attributes[edgeMetric] !== undefined) {
+				const hasFromNodeEdgePreview = edgePreviewNodes.has(edge.fromNodeName)
+				const hasToNodeEdgePreview = edgePreviewNodes.has(edge.toNodeName)
+				if (hasFromNodeEdgePreview !== hasToNodeEdgePreview) {
+					if (hasFromNodeEdgePreview) {
+						edge.visible = EdgeVisibility.from
+					} else {
+						edge.visible = EdgeVisibility.to
+					}
+				} else if (hasFromNodeEdgePreview) {
+					edge.visible = EdgeVisibility.both
+				} else {
+					edge.visible = EdgeVisibility.none
 				}
 			} else {
 				edge.visible = EdgeVisibility.none
 			}
```

Comment:
I think we can omit two else-branches by setting EdgeVisibility.none as a default value.
```suggestion
		for (const edge of edges) {
		        edge.visible = EdgeVisibility.none

			if (edge.attributes[edgeMetric] !== undefined) {
				const hasFromNodeEdgePreview = edgePreviewNodes.has(edge.fromNodeName)
				const hasToNodeEdgePreview = edgePreviewNodes.has(edge.toNodeName)
				if (hasFromNodeEdgePreview !== hasToNodeEdgePreview) {
					if (hasFromNodeEdgePreview) {
						edge.visible = EdgeVisibility.from
					} else {
						edge.visible = EdgeVisibility.to
					}
				} else if (hasFromNodeEdgePreview) {
					edge.visible = EdgeVisibility.both
				}
			}
```